[BREAKING CHANGE] Create new `Select` with `react-select@3` (#2158)
KissKissBankBank/kitten: assets/javascripts/kitten/components/form/select-with-state/select.stories.mdx

import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks'
import * as stories from './select.stories.js'
import { SelectWithState } from './index'

<Meta title="Form/SelectWithState" component={SelectWithState} />

# SelectWithState

---

Our select is based on `react-select` library https://react-select.com/ (v3.1).

## Default

<Canvas>
  <Story name="Default" story={stories.Default} />
</Canvas>

## With default value

> Our select is `uncontrolled`, the `value` behave as **default value**, usefull to link it with form lib (Formik…).

Reviewer: ```suggestion
Our select is `uncontrolled`: the `value` behave as **default value**, it is useful when linking it with form libraries (Formik…).
```

<Canvas>
  <Story name="WithDefaultValue" story={stories.WithDefaultValue} />
</Canvas>

## With custom styles

> We extend default styles using `classNames`, follow this section to know how to : https://react-select.com/styles#using-classnames. Our classNames prefix is `k-Select`.

Reviewer: ```suggestion
We extend default styles using `classNames`, this help section show you how to: https://react-select.com/styles#using-classnames. Our classNames prefix is `k-Select`.
```
